modules/PhenoDigm.py

```diff
@@ -234,53 +249,104 @@ def load_data_from_cache(self):
+            self.load_tsv(
+                self.MGI_PUBMED_FILENAME,
+                column_names=['_0', '_1', '_2', 'mp_id', 'literature', 'targetInModelMgiId']
+            )
+            .select('mp_id', 'literature', 'targetInModelMgiId')
+            .distinct()
+            # Separate and explode targets.
+            .withColumn('targetInModelMgiId', pf.split(pf.col('targetInModelMgiId'), r'\|'))
+            .withColumn('targetInModelMgiId', pf.explode('targetInModelMgiId'))
+            # Separate and explode literature references.
+            .withColumn('literature', pf.split(pf.col('literature'), r'\|'))
+            .withColumn('literature', pf.explode('literature'))
+            .select('mp_id', 'literature', 'targetInModelMgiId')
+            # E.g. 'MP:0000600', '12529408', 'MGI:97874'.
         )
-        disease_human_phenotypes = (
-            self.disease
-            .withColumn('hp_id', pf.expr(r"regexp_extract_all(disease_phenotypes, '(HP:\\d+)', 1)"))
-            .withColumn('hp_id', pf.explode('hp_id'))
-            .select('disease_id', 'hp_id')
+        # Literature references for a given (model, gene) combination.
+        self.literature = (
+            self.disease_model_summary
+            .select('model_id', 'targetInModelMgiId')
+            .distinct()
+            .join(self.model_mouse_phenotypes, on='model_id', how='inner')
+            .join(mgi_pubmed, on=['targetInModelMgiId', 'mp_id'], how='inner')
+            .groupby('model_id', 'targetInModelMgiId')
+            .agg(pf.collect_set(pf.col('literature')).alias('literature'))
+            .select('model_id', 'targetInModelMgiId', 'literature')
         )
+
+    @staticmethod
+    def _cleanup_model_identifier(dataset):
+        return (
+            # Model ID adjustments. First, strip the trailing modifiers, where present. The original ID, used for table
+            # joins, may look like 'MGI:6274930#hom#early', where the first part is the allele ID and the second
+            # specifies the zygotic state. There can be several models for the same allele ID with different phenotypes.
+            # However, this information is also duplicated in `biologicalModelGeneticBackground` (for example:
+            # 'C57BL/6NCrl,Ubl7<em1(IMPC)Tcp> hom early'), so in this field we strip those modifiers.
+            dataset
+            .withColumn(
+                'biologicalModelId',
+                pf.split(pf.col('model_id'), '#').getItem(0)
+            )
+            .drop('model_id')
+
+            # Second, we only want to output the model names from the MGI namespace. An example of something we *don't*
+            # want is 'NOT-RELEASED-025eb4a791'. This will be converted to null.
+            .withColumn(
+                'biologicalModelId',
+                pf.when(pf.col('biologicalModelId').rlike(r'^MGI:\d+$'), pf.col('biologicalModelId'))
+            )
+        )
+
+    def generate_phenodigm_evidence_strings(self, score_cutoff):
```

Comment:
Yes, currently this is the case. I believe @ireneisdoomed had some thoughts & back of the envelope calculations for some sensible threshold values which we could pick, so that the results are broadly comparable to the previous ones.